### DISCUSSION TITLE
Is gevent's cooperative multitasking needed for a uvicorn ASGI server?

### DISCUSSION THREAD

**Discussion by chrisinmtown** [spec-first/connexion#1994] [Q&A]
My Connexion V2 app used the following lines in `__init__.py` to allow multiple requests to be processed in parallel:
```
# configure libraries for cooperative multitasking via
# gevent which allows concurrent request processing
import gevent.monkey; gevent.monkey.patch_all()  # noqa: E702
```
For example, if one request has to perform a big messy database join, then another request (or 100) could be processed while that first one is cooking along.

Please comment, what's the right way to enable similar behavior in Connexion V3? I see this note in the migration guide:

> Concurrency in the middleware stack is handled by the async event loop, but once a request is passed to the underlying Flask app, it is executed in a thread pool (of 10 workers) automatically.

Does this mean as many as 10 requests will automagically be processed in parallel, even without monkey-patching in any `gevent` cooperative multi-tasking code?

Thanks in advance.

**Comment by RobbeSneyders** [comment]
Hi @chrisinmtown,

Your code will still need to be cooperative to benefit from the threads. Note that Python threads run concurrently, not in parallel. One way to make your code cooperative is by using `gevent.monkey.patch_all()`, although this is mainly meant to make third party code cooperative. For code that is in your control, leveraging the `AsyncApp` would be recommended if you're not highly dependent on Flask.

**Reply by chrisinmtown** [reply]
Thanks for the quick reply. I'm glad to hear we can continue using the cooperative multitasking approach.  

I have been struggling to answer the question if my apps are "highly dependent on Flask."  The apps /definitely/ use the Flask configuration object, it's just handy to populate from an environment file that points to a file. What are other examples of Flask features that may have been built in when I might not have been paying close attention?

**Reply by RobbeSneyders** [reply]
Mostly dependency on the Flask ecosystem (Eg. extensions) I would say, although some of those no longer work with Connexion 3, and should be replaced with ASGI or WSGI middleware anyway.

**Comment by chrisinmtown** [comment]
It doesn't appear to work tho, I started a new discussion: https://github.com/spec-first/connexion/discussions/1998
